examples/http/src/main/java/io/opentelemetry/example/HttpServer.java

```diff
@@ -0,0 +1,165 @@
+/*
+ * Copyright 2019, OpenTelemetry Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.opentelemetry.example;
+
+import com.sun.net.httpserver.HttpExchange;
+import com.sun.net.httpserver.HttpHandler;
+import io.opentelemetry.context.propagation.HttpTextFormat;
+import io.opentelemetry.exporters.inmemory.InMemorySpanExporter;
+import io.opentelemetry.sdk.OpenTelemetrySdk;
+import io.opentelemetry.sdk.trace.SpanData;
+import io.opentelemetry.sdk.trace.TracerSdkFactory;
+import io.opentelemetry.sdk.trace.export.SimpleSpansProcessor;
+import io.opentelemetry.trace.AttributeValue;
+import io.opentelemetry.trace.Span;
+import io.opentelemetry.trace.SpanContext;
+import io.opentelemetry.trace.Tracer;
+import io.opentelemetry.trace.propagation.HttpTraceContext;
+
+import java.io.IOException;
+import java.io.OutputStream;
+import java.net.InetSocketAddress;
+import java.util.HashMap;
+import java.util.Map;
+
+public class HttpServer {
+
+  private class helloHandler implements HttpHandler {
+
+    @Override
+    public void handle(HttpExchange he) throws IOException {
+      Span span = tracer.spanBuilder("hello handler").startSpan();
+      try {
+        // Extract the context from the HTTP request
+        SpanContext ctx = extractCtx(he);
+        // Rebuild a span with the received context
+        span = tracer.spanBuilder("hello handler").setParent(ctx).startSpan();
+
+        // Process the request
+        answer(he, span);
+      } catch (StringIndexOutOfBoundsException e) {
+        // msg without ctx
+      } catch (Exception e) {
+        // catch other errors
+        System.out.println(e.getMessage());
+      } finally {
+        // Close the span
+        span.end();
+      }
+    }
+
+    private SpanContext extractCtx(HttpExchange he) {
+      SpanContext ctx = (SpanContext) textFormat.extract(he, getter);
```

Comment:
I'd do that right in `handle()` for this example since it's pretty concise anyway.